extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/common/SortMergeJoinFrameProcessor.java

```diff
@@ -533,6 +629,11 @@ public long totalBytesBuffered()
       return bytes;
     }
 
+    public boolean canBufferMoreFrames()
```

Comment:
Sure thing. Added this:

```
    /**
     * Whether this tracker can accept more frames without exceeding {@link #maxBufferedBytes}. Always returns true
     * if the number of buffered frames is zero or one, because the join algorithm may require two frames being
     * buffered. (For example, if we need to verify that the last row in a frame contains a complete set of a key.)
     */
```